tests/templates/test_subroutines/test_qrom.py

```diff
@@ -258,3 +272,12 @@ def test_wrong_wires_error(bitstrings, control_wires, target_wires, msg_match):
     """Test that error is raised if more ops are requested than can fit in control wires"""
     with pytest.raises(ValueError, match=msg_match):
         qml.QROM(bitstrings, control_wires, target_wires, work_wires=None)
+
+
+def test_none_work_wires_case():
+    """Test that clean version is not applied if work wires are not used"""
+
+    gates = qml.QROM.compute_decomposition(["1", "0", "0", "1"], [0, 1], [2], [], clean=True)
+
+    for gate in gates:
+        assert gate.name != "Hadamard"
```

Comment:
Haha, I see what you are doing here, but I would say this is a "lazy"-test. Since it's only one test case, I would say compute the expected decomposition by hand, and then test that the compute decomposition matches it. 

Something like: 
```suggestion
def test_none_work_wires_case():
    """Test that clean version is not applied if work wires are not used"""

    gates_clean = qml.QROM.compute_decomposition(["1", "0", "0", "1"], [0, 1], [2], [], clean=True)
    expected_gates = qml.QROM.compute_decomposition(["1", "0", "0", "1"], [0, 1], [2], [], clean=False)
    
    assert gates_clean == expected_gates
```